pandas/tests/dtypes/test_concat.py

```diff
@@ -49,3 +49,13 @@ def test_concat_periodarray_2d():
 
     with pytest.raises(ValueError, match=msg):
         _concat.concat_compat([arr[:2], arr[2:]], axis=1)
+
+
+def test_concat_series_between_empty_and_tzaware_series():
+    tzaware_time = pd.Timestamp("2020-01-01T00:00:00+00:00")
+    ser1 = Series(index=[tzaware_time], data=0, dtype=float)
+    ser2 = Series(dtype=float)
+
+    result = pd.concat([ser1, ser2], axis=1)
+    tm.assert_index_equal(pd.Index([tzaware_time], dtype="object"), result.index)
+    tm.assert_series_equal(Series([0.0, None], name=tzaware_time), result.iloc[0])
```

Comment:
Could you construct an `expected = pd.DataFrame(...)` and use `tm.assert_frame_equal(result, expected)`?

Reply:
Sure. Just changed the format